src/Http/Uri.php

```diff
@@ -21,37 +21,17 @@
  */
 class Uri implements UriInterface
 {
-    /** @var string */
-    private $query = '';
-
-    /** @var string */
-    private $scheme = '';
-
-    /** @var string */
-    private $host = '';
-
-    /** @var string */
-    private $user = '';
-
-    /** @var string */
-    private $password = '';
-
-    /** @var int|null */
-    private $port;
-
-    /** @var string */
-    private $path = '';
-
-    /** @var string */
-    private $fragment = '';
-
-    /** @var string */
+    private string $query = '';
+    private string $scheme = '';
+    private string $host = '';
+    private string $user = '';
+    private string $password = '';
+    private ?int $port;
```

Comment:
```suggestion
    private ?int $port = null;
```
wenn `$uri` im Constructor gleich `''` ist, wird die Property niemals gesetzt, sodass ein Aufruf von `getPort` mit einem `TypeError` fehlschlagen wird

Reply:
Das wird ja im Konstruktor über die "filterPort" Methode abgefangen.